src/h3lib/lib/localij.c

```diff
@@ -204,7 +204,7 @@ H3Error cellToLocalIjk(H3Index origin, H3Index h3, CoordIJK *out) {
                 // TODO: We may be unfolding the pentagon incorrectly in this
                 // case; return an error code until this is guaranteed to be
                 // correct.
-                return E_FAILED;
+                return E_PENTAGON;  // invalid inputs or algo improvements needed?
```

Comment:
I feel like this needs a different error, e.g. `E_UNREACHABLE_CELL` (could replace `E_TOO_FAR`). In other contexts, `E_PENTAGON` means we hit a pentagon and the algo can't handle it - in this case it's more specific, only certain directions around a pentagon are unreachable, and pentagons themselves are not failure cases.

Reply:
What would you think about breaking `E_PENTAGON` into two different errors, to express either:

1. We hit a pentagon, and the math was tricky, so there might be a "safe" function you can use to recover from this, or such a function may get implemented in the future
2. We hit a pentagon, but something is actually wrong here and this error is not recoverable

Would either of those be covered by what you were thinking with `E_UNREACHABLE_CELL`?

Reply:
See my other comment - in this case, the pentagon is an unnecessary detail to the main issue: We can't reach the cell you want. I don't want to expose the exact reasons here, because they are not fundamental to the grid, only to our partial implementation. All the user needs/wants to know is that some cases aren't handled, and this is one of them - we actually don't want to encourage them to try to find weird workarounds, because they would be totally dependent on our current, flawed implementation.

Reply:
https://github.com/uber/h3/issues/627